libs/utils/android/workloads/pcmark.py

```diff
@@ -0,0 +1,182 @@
+import os
+import logging
+import time
+
+from subprocess import Popen, PIPE
+
+from android import Screen, System
+from android.workload import Workload
+
+PCMARK_DB_FILE = 'pcmarkscores.txt'
+PCMARK_LOG_FILE = 'pcmarklog.txt'
+PCMARK_TESTS = ['PERFORMANCE', 'BATTERY_LIFE']
+
+# Regexps for benchmark synchronization
+PCMARK_BENCHMARK_START_RE = re.compile(
+    r'com.futuremark.pcmandroid.AbstractMainActivity: '\
+    'running benchmark'
+)
+
+PCMARK_BENCHMARK_END_RE = re.compile(
+    r'com.futuremark.pcmandroid.AbstractMainActivity: '\
+    'onWebViewReady.*view_scoredetails.html'
+)
+
+class PCMark(Workload):
+    """
+    Android PCMark workload
+    """
+
+    # Package required by this workload
+    package = 'com.futuremark.pcmark.android.benchmark'
+
+    def __init__(self, test_env):
+        super(PCMark, self).__init__(test_env)
+        self._log = logging.getLogger('PCMark')
+        self._log.debug('Workload created')
+
+    def run(self, out_dir, test_name, collect=''):
+        """
+        Run single PCMark workload. Returns a collection of results.
+
+        :param out_dir: Path to experiment directory on the host
+                        where to store results.
+        :type out_dir: str
+
+        :param test_name: Name of the test to run
+        :type test_name: str
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+            - 'systrace'
+            - 'ftrace'
+            - any combination of the above as a single space-separated string.
+        :type collect: list(str)
+        """
+
+        self.out_dir = out_dir
+        self.collect = collect
+
+        # Unlock device screen (assume no password required)
+        System.menu(self._target)
+        # Make sure we exit the app if already open
+        System.back(self._target)
+
+        # Close and NOT clear application (benchmark tests are downloaded from web)
+        System.force_stop(self._target, self.package, clear=False)
+
+        # Set airplane mode
+        System.set_airplane_mode(self._target, on=True)
+
+        # Set min brightness
+        Screen.set_brightness(self._target, auto=False, percent=0)
+
+        # Force screen in PORTRAIT mode
+        Screen.set_orientation(self._target, portrait=True)
+
+        # Clear logcat
+        self._target.clear_logcat()
+
+        # Start benchmark
+        System.monkey(self._target, self.package)
+        # Wait a few seconds while application is being loaded
+        time.sleep(8)
+
+        # Parse logcat ActivityManager output lines to detect start
+        logcat_cmd = self._adb(
+                'logcat com.futuremark.pcmandroid.AbstractMainActivity:I *:S'\
+                .format(self._target.adb_name))
+        self._log.info("%s", logcat_cmd)
+
+        logcat = Popen(logcat_cmd, shell=True, stdout=PIPE)
+
+        # Run performance workload
+        if test_name.upper() == 'PERFORMANCE':
+            self._target.execute('input tap 750 1450')
```

Comment:
Here we should use relative coords... but it's a pain.

On a Pixel XL, where the screen is 2k (1440,2560), the run button has this coords:
```
   top-left:     ( 914,1630)
   bottom-right: (1340,1730)
```